vhdx/src/vhdx_header.rs

```diff
@@ -0,0 +1,552 @@
+        let mut header2_seq_num: u64 = 0;
+        let mut no_valid_header: bool = true;
+
+        if let Ok(ref header_1) = header_1 {
+            no_valid_header = false;
+            header1_seq_num = header_1.sequence_number;
+        }
+
+        if let Ok(ref header_2) = header_2 {
+            no_valid_header = false;
+            header2_seq_num = header_2.sequence_number;
+        }
+
+        if no_valid_header {
+            panic!("Get current header: {}", Error::NoValidHeader);
+        } else if header1_seq_num >= header2_seq_num {
+            (1, header_1, header_2)
+        } else {
+            (2, header_1, header_2)
+        }
+    }
+
+    /// _update_header() takes two header and update the noncurrent header with
+    /// the corrent one. Returns both headers as a tuple sequenced as it received
+    /// them in the parameter list.
+
+    fn _update_header(
+        f: &mut File,
+        header_1: Result<Header>,
+        header_2: Result<Header>,
+        guid: u128,
+    ) -> (Header, Header) {
+        let (current_header, header_1, header_2) =
+            VhdxHeader::get_current_header(header_1, header_2);
+        if current_header == 1 {
+            match header_1 {
+                Ok(header_1) => {
+                    let header_2 =
+                        Header::update_header(f, header_1.clone(), true, guid, HEADER2_START);
+                    (header_1, header_2)
+                }
+                Err(e) => {
+                    panic!("Header 2 update failed{}", e);
+                }
+            }
+        } else {
+            match header_2 {
+                Ok(header_2) => {
+                    let header_1 =
+                        Header::update_header(f, header_2.clone(), true, guid, HEADER2_START);
+                    (header_1, header_2)
+                }
+                Err(e) => {
+                    panic!("Header 1 update failed{}", e);
+                }
+            }
+        }
+    }
+
+    fn update_headers(
+        f: &mut File,
+        header_1: Result<Header>,
+        header_2: Result<Header>,
+        guid: u128,
+    ) -> (Header, Header) {
+        // According to the spec, update twice
+        let (header_1, header_2) = VhdxHeader::_update_header(f, header_1, header_2, guid);
+        VhdxHeader::_update_header(f, Ok(header_1), Ok(header_2), guid)
+    }
+
+    pub fn get_region_entry_count(self) -> u32 {
+        return self.region_table_1.entry_count;
+    }
+}
+
+pub fn calculate_checksum(buffer: &mut [u8], csum_offset: usize) -> Result<u32> {
+    let orig_csum: u32;
+    let zero_bytes = [0; 4];
+
+    // Read the checksum in to a mutable slice
+    let mut csum_buf = &mut buffer[csum_offset..csum_offset + 4];
+    // Conver the checksum chunk in to a u32 integer
+    orig_csum = LittleEndian::read_u32(&mut csum_buf);
+    // Zeroed the checksum in the buffer. should we use Zeroize?
+    LittleEndian::write_u32_into(&zero_bytes, &mut csum_buf);
```

Comment:
Here you should pass the buffer directly (as a reference):
```suggestion
    LittleEndian::write_u32_into(&zero_bytes, buffer);
```